### PR TITLE
fix: remove details and summary tag on wwe (fix #686)

### LINKED ISSUE #686: The <details /> and <summary /> are removed from WYSIWYG
## Version

v1.4.7

## Test Environment

All

## Current Behavior

1. Disable the `useDefaultHTMLSanitizer` option.
```js
const editor = new tui.Editor({
  // ... 
  useDefaultHTMLSanitizer: false // default: true
});
```
2. When entering html in Markdown, enter the [`details`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) and [`summary`](https://developer.mozilla.org/bn/docs/Web/HTML/Element/summary) tags.
3. The `details` and `summary` tags are rendered in the Markdown preview, but when switching to WYSIWYG, these tags are removed.

## Cause

Remove elements except block elements allowed in Squire.
```js
var allowedBlock = /^(?:A(?:DDRESS|RTICLE|SIDE|UDIO)|BLOCKQUOTE|CAPTION|D(?:[DLT]|IV)|F(?:IGURE|IGCAPTION|OOTER)|H[1-6]|HEADER|HR|L(?:ABEL|EGEND|I)|O(?:L|UTPUT)|P(?:RE)?|SECTION|T(?:ABLE|BODY|D|FOOT|H|HEAD|R)|COL(?:GROUP)?|UL)$/;
```

## Expected Behavior

The `details` and `summary` tags should render correctly in WYSIWYG.

> Notice: When using the `details` and `summary` tags in the editor,  these tags are not fully supported by WYSIWYG

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

* Issue related to #686

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/wysiwygEditor.js [hunk 1]
```diff
@@ -1182,13 +1184,14 @@ class WysiwygEditor {
...
    *     @param {boolean} [options.useCommandShortcut=true] - whether to use squire command shortcuts
+   *     @param {boolean} [options.useDefaultHTMLSanitizer=true] - whether to use default html sanitizer
    * @returns {WysiwygEditor} wysiwygEditor
    * @ignore
    */
   static factory($el, eventManager, options) {
     const wwe = new WysiwygEditor($el, eventManager, options);
 
-    wwe.init();
+    wwe.init(options.useDefaultHTMLSanitizer);
```
Comment (dongwoo-kim) [inline src/js/wysiwygEditor.js]: init 함수가 받는 값은 정상적인 boolean 값이면 좋을 것 같아요. 요런 식으로? 
```js
wwe.init(options.useDefaultHTMLSanitizer !== false);
```

File: src/js/editor.js [hunk 2]
```diff
@@ -216,7 +216,7 @@ class ToastUIEditor {
       this.convertor,
       false,
       this.options.previewDelayTime);
-    this.wwEditor = WysiwygEditor.factory(this.layout.getWwEditorContainerEl(), this.eventManager);
+    this.wwEditor = WysiwygEditor.factory(this.layout.getWwEditorContainerEl(), this.eventManager, this.options);
```
Comment (dongwoo-kim) [inline src/js/editor.js]: 통째로 넘기지 말고, wysiwyg 에서 사용하는 값만 추려서 넘기는 게 낫지 낳을까요?

Comment (seonim-ryu) [reply src/js/editor.js]: @dongwoo-kim
이런식으로 말씀하시는거죠?
```js
WysiwygEditor.factory(this.layout.getWwEditorContainerEl(), this.eventManager, {
  useDefaultHTMLSanitizer: this.options. useDefaultHTMLSanitizer
});
```

File: src/js/wysiwygEditor.js [hunk 3]
```diff
@@ -72,7 +73,8 @@ class WysiwygEditor {
       blockTag: 'DIV',
       leafNodeNames: {
         'HR': false
-      }
+      },
+      allowedBlocks: useDefaultHTMLSanitizer === false ? ['details', 'summary'] : []
```
Comment (seonim-ryu) [inline src/js/wysiwygEditor.js]: `useDefaultHTMLSanitizer ? [] : ['details', 'summary']`

File: src/js/wysiwygEditor.js (hunk 1 above)
Comment (seonim-ryu) [reply src/js/wysiwygEditor.js]: Line#77 참조

**Pull request merged**